### DISCUSSION TITLE
How to obtain the initial position of a crazyflie using the lighthouse positioning system

### DISCUSSION THREAD

**Discussion by Morgens2** [bitcraze/discussions#1438] [Q&A] [answered]
Hi there!
I am using 3 Crazyflie drones and a Lighthouse positioning system. I wonder if there is a way to obtain the initial position of each drone, before they take off, in order to use it as a constant value later on. Can someone please help me it figure out? Thank you in advance.

**Comment by knmcguire** [comment, marked as answer]
Hi! You can try get_estimated_positions() from the swarm class? https://www.bitcraze.io/documentation/repository/crazyflie-lib-python/master/api/cflib/crazyflie/swarm/ and then assigning these to perhaps some global python variables based on the drone name. 

Mind that this function is meant to only be used ones and not in a loop.

**Reply by Morgens2** [reply]
Hi! Thanks for your fast response! get_estimated_positions() is indeed the function I need since I want to get the positions once. However, after calling it in main it doesn't return any data. Is this normal? Also, how can I get the position of each URI from the returned dict?
Thanks for your answers in advance.
Here is the main function I used.
```python
if __name__ == '__main__':
    cflib.crtp.init_drivers()
    factory = CachedCfFactory(rw_cache='./cache')
    with Swarm(uris, factory=factory) as swarm:
        print('Connected to  Crazyflies')
        swarm.reset_estimators()
        time.sleep(1)
        swarm.get_estimated_positions()
```

**Reply by knmcguire** [reply]
The function returns a dict, so you'll need to assign a variable to it like `dict_pos = swarm.get_estimated_positions()`

The rest I'm afraid is very python specific, so some googling is required.

**Reply by Morgens2** [reply]
Great! Thanks for your time.
